libraries/Microsoft.Bot.Builder.Dialogs/Choices/Channel.cs

```diff
@@ -1,79 +1,85 @@
-                case Connector.Channels.Skype:
-                case Connector.Channels.Msteams:
-                    return buttonCnt <= 3;
-
-                case Connector.Channels.Slack:
-                case Connector.Channels.Emulator:
-                case Connector.Channels.Directline:
-                case Connector.Channels.Webchat:
-                case Connector.Channels.Cortana:
-                    return buttonCnt <= 100;
-
-                default:
-                    return false;
-            }
-        }
-
-        public static bool HasMessageFeed(string channelId)
-        {
-            switch (channelId)
-            {
-                case Connector.Channels.Cortana:
-                    return false;
-
-                default:
-                    return true;
-            }
-        }
-
-        public static int MaxActionTitleLength(string channelId) => 20;
-
-        public static string GetChannelId(ITurnContext turnContext) => string.IsNullOrEmpty(turnContext.Activity.ChannelId)
-            ? string.Empty : turnContext.Activity.ChannelId;
-
-        // This class has been deprecated in favor of the class in Microsoft.Bot.Connector.Channels located
-        // at https://github.com/Microsoft/botbuilder-dotnet/libraries/Microsoft.Bot.Connector/Channels.cs.
-        // This change is non-breaking and this class now inherits from the class in the connector library.
-        [Obsolete("This class is deprecated. Please use Microsoft.Bot.Connector.Channels.")]
-        public class Channels : Connector.Channels
-        {
-        }
-    }
-}
+using System;
+
+namespace Microsoft.Bot.Builder.Dialogs.Choices
+{
+    public class Channel
+    {
+        public static bool SupportsSuggestedActions(string channelId, int buttonCnt = 100)
+        {
+            switch (channelId)
+            {
+                case Connector.Channels.Facebook:
+                case Connector.Channels.Skype:
+                    return buttonCnt <= 10;
+
+                case Connector.Channels.Line:
+                    return buttonCnt <= 13;
+
+                case Connector.Channels.Kik:
+                    return buttonCnt <= 20;
+
+                case Connector.Channels.Slack:
+                case Connector.Channels.Telegram:
+                case Connector.Channels.Emulator:
+                case Connector.Channels.Directline:
+                case Connector.Channels.Webchat:
+                    return buttonCnt <= 100;
+
+                default:
+                    return false;
+            }
+        }
+
+        public static bool SupportsCardActions(string channelId, int buttonCnt = 100)
+        {
+            switch (channelId)
+            {
+                case Connector.Channels.Facebook:
+                case Connector.Channels.Skype:
+                case Connector.Channels.Msteams:
+                    return buttonCnt <= 3;
+
+                case Connector.Channels.Line:
```

Comment:
Same comment - please include a link (or text) explaining where this value comes from.

Reply:
I could find no docs stating Line supports 99 card actions, though I tried this on Line for Windows and Android.